test/11_array.jl

```diff
@@ -136,6 +136,14 @@ using CategoricalArrays: DefaultRefType, catvaluetype, leveltype
     @test levels(x2) !== levels(x)
     @test isordered(x2) == isordered(x)
 
+    x2 = copy(x)
+    @test fill!(x2, "a") === x2
+    @test x2 == ["a", "a", "a"]
+    @test levels(x2) == ["a", "b"]
+    fill!(x2, "c")
```

Comment:
Would make sense to use e.g. "0" here (or use "a" but use "b" and "c" above) so that you can distinguish between the new level being added to the end of the list and the new level being added in sorted order.